server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,287 @@
+            .equals(joiningNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY))) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY
+                    + "] different than existing node ["
+                    + existingNode
+                    + "]"
+            );
+        }
+        if (!existingNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY)
+            .equals(joiningNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY))) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY
+                    + "] different than existing node ["
+                    + existingNode
+                    + "]"
+            );
+        }
+        if (!existingNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY)
+            .equals(joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY))) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY
+                    + "] different than existing node ["
+                    + existingNode
+                    + "]"
+            );
+        }
+        if (!existingNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY)
+            .equals(joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY))) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY
+                    + "] different than existing node ["
+                    + existingNode
+                    + "]"
+            );
+        }
+        if (!existingNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY)
+            .equals(joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY))) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY
+                    + "] different than existing node ["
+                    + existingNode
+                    + "]"
+            );
+        }
+    }
+
+    private static Settings buildSettings(String stringSettings) {
```

Comment:
Lets check if we can use the following method - 
https://github.com/opensearch-project/OpenSearch/blob/5a25350e7e2f4224d50744a8fb1def65b71f0356/server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java#L138-L141

Like @gbbafna if the settings are nested, this can easily break. It would be good to use a programmatic deserialisation approach here.